addons/api/addon/utils/flatten-nested-object.js

```diff
@@ -0,0 +1,29 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+
+/**
+ * Flattens a nested object into a single-level object.
+ * @param obj
+ * @param result
+ * @returns {Object}
+ */
+
+export const flattenObject = (obj, result = {}) => {
+  for (const key in obj) {
+    // Skip the 'metadata' key
+    if (key === 'metadata') {
+      continue;
+    }
+    // Check if the value is an object and it is not null
+    if (typeof obj[key] === 'object' && obj[key] !== null) {
+      // Recursively flatten the object
+      flattenObject(obj[key], result);
+    } else if (obj[key]) {
+      // Only add to result if the value is not null
+      result[key] = obj[key];
```

Comment:
What happens if you have nested keys that are the same name?

Reply:
ohh yeah it will overwrite the previous entry. I will have to preserve the path of the child obj. I will add back the path logic I originally had

Reply:
but I wonder if it's a good UX to show the `parentKey.key` as the field name 🤔  But if we want to preserve uniqueness and avoid being overwritten then we might have to go with this approach